ignite/cmd/bubblemodel/testnet_multi_node.go

```diff
@@ -0,0 +1,233 @@
+					}
+				}
+			}()
+			return UpdateStatusMsg{nodeIdx: nodeIdx, status: Running}
+		}
+		// Use kill to stop the node process by PID
+		if *pid != 0 {
+			err := syscall.Kill(-*pid, syscall.SIGTERM) // Stop the daemon process
+			if err != nil {
+				fmt.Printf("Failed to stop node %d: %v\n", nodeIdx+1, err)
+			} else {
+				*pid = 0 // Reset PID after stopping
+			}
+		}
+		return UpdateStatusMsg{nodeIdx: nodeIdx, status: Stopped}
+	}
+}
+
+// StopAllNodes stops all nodes.
+func (m *MultiNode) StopAllNodes() {
+	for i := 0; i < m.numNodes; i++ {
+		if m.nodeStatuses[i] == Running {
+			RunNode(i, false, *m)() // Stop node
+		}
+	}
+}
+
+// Update handles messages and updates the model.
+func (m MultiNode) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "q":
+			m.StopAllNodes() // Stop all nodes before quitting
+			return m, tea.Quit
+		default:
+			// Check for numbers from 1 to numNodes
+			for i := 0; i < m.numNodes; i++ {
+				if msg.String() == fmt.Sprintf("%d", i+1) {
+					return m, ToggleNode(i)
+				}
+			}
+		}
+
+	case ToggleNodeMsg:
+		if m.nodeStatuses[msg.nodeIdx] == Running {
+			return m, RunNode(msg.nodeIdx, false, m) // Stop node
+		}
+		return m, RunNode(msg.nodeIdx, true, m) // Start node
+
+	case UpdateStatusMsg:
+		m.nodeStatuses[msg.nodeIdx] = msg.status
+		return m, UpdateDeemon()
+	case UpdateLogsMsg:
+		return m, UpdateDeemon()
+	}
+
+	return m, nil
+}
+
+// View renders the interface.
+func (m MultiNode) View() string {
+	// Define styles for the state
+	runningStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("2"))                               // green
+	stoppedStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("1"))                               // red
+	tcpStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("3"))                                   // yellow
+	grayStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("8"))                                  // gray
+	purpleStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("5"))                                // purple
+	statusBarStyle := lipgloss.NewStyle().Background(lipgloss.Color("0"))                             // Status bar style
+	blueStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("45")).Background(lipgloss.Color("0")) //blue
+
+	statusBar := blueStyle.Render("Press q to quit | Press 1-4 to ") + statusBarStyle.Render(runningStyle.Render("start")) + blueStyle.Render("/") + statusBarStyle.Render(stoppedStyle.Render("stop")) + blueStyle.Render(" corresponding node")
+	output := statusBar + "\n\n"
+
+	// Add node control section
+	output += purpleStyle.Render("Node Control:")
+	for i := 0; i < m.numNodes; i++ {
+		status := stoppedStyle.Render("[Stopped]")
+		if m.nodeStatuses[i] == Running {
+			status = runningStyle.Render("[Running]")
+		}
+
+		tcpAddress := tcpStyle.Render(fmt.Sprintf("tcp://127.0.0.1:%d", 26657-3*i))
```

Comment:
we have a pkg to find the available port

https://github.com/ignite/cli/blob/d79a76dd56026c45ea072ada2ecbc34cb9a2103f/ignite/pkg/availableport/availableport.go#L24-L24

Reply:
The possibility of duplicate ports is very low, can I just fix and avoid the default ports?

Reply:
Add the pkg is more easy to avoid the default ports

Reply:
ok, i will find out and use the port from the package you suggested.